### ISSUE TITLE
IDE0027: false positive with pragmas

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn#68913]
**Version Used**: dotnet 8.0.100-preview.5.23303.2

**Steps to Reproduce**:

```csharp
public class C
{
    public void M()
	{
#if NETCOREAPP
		// some code
#endif

		Console.WriteLine("ok"); // IDE0027 raised when in non-netcoreapp
	}
}
```

**Diagnostic Id**: IDE0027 - expression body

**Expected Behavior**: Not flagged since in one compilation we cannot use expression body

**Actual Behavior**:

**Comment by mavasani** [comment]
This should a trivial fix, let me create a PR later today.

**Comment by CyrusNajmabadi** [comment]
Closing out as i do not repro this.

**Comment by Evangelink** [comment]
I confirm that I cannot reproduce anymore.
